libsql-storage-server/src/service.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::atomic::AtomicU32;
+use std::sync::Arc;
+
+use crate::memory_store::InMemFrameStore;
+use crate::store::FrameStore;
+use libsql_storage::rpc;
+use libsql_storage::rpc::storage_server::Storage;
+use tokio::sync::Mutex;
+use tonic::{Request, Response, Status};
+use tracing::{error, trace};
+
+pub struct Service {
+    store: Arc<Mutex<dyn FrameStore + Send + Sync>>,
```

Comment:
dont need to arc tonic  will do that for you. It also feels like this locking should happen at the store layer rather than here. Clients that go over the network in general don't need a mutex.